### ISSUE TITLE
Create public C8 REST API guidelines

### ISSUE THREAD

**Issue by tmetzke** [camunda/camunda#24239]
## Description

- [x] Provide a public version of the (🔒 internal) [C8 REST API guidelines](https://docs.google.com/document/d/1G9AmmNac-4QLGZ0LXQXa3FyeCrSJdIboaPt3-R6dWNw/edit?tab=t.0)
- [x] Replace the internal content with the public reference as much as possible (ideally everything in [API guidelines](https://docs.google.com/document/d/1G9AmmNac-4QLGZ0LXQXa3FyeCrSJdIboaPt3-R6dWNw/edit?tab=t.0#heading=h.2spfyxzbwgko))

**Comment by akeller** [comment]
@tmetzke, I mentioned this on the docs PR, but who is the audience for this? Camundi, customers, or both? When I reviewed the PR, it was unclear.

**Comment by tmetzke** [comment]
> @tmetzke, I mentioned this on the docs PR, but who is the audience for this? Camundi, customers, or both? When I reviewed the PR, it was unclear.

Yep, it's on my (completely crowded) radar 😅  The audience should be users in general to understand how we design our API. It's currently too focused on Camundi because it's just a copy of the internal guidelines.

**Comment by tmetzke** [comment]
I adjusted the internal guidelines with references to the now-public documentation.
